### PR TITLE
Restructure Vault.md into HardwareWalletIntegration.md

### PR DESCRIPTION
**Pull request by pavlenex**
Closes: 

- Rename doc to HardwareWalletIntegration.md
- Reduced the unnecessary mentioning of Vault.md as it's irrelevant to the user
- Emphasizes the HWI and why it's important
-  Takes care of redirects /Vault to /HardwareWalletIntegration

Needs some local testing to make sure everything works and please review for clarity and let me know if I missed anything or if anything requires further.

### CODE REVIEW

**Comment by pavlenex** [comment]
The linkcheck reporting 404 that aren't 404 nor related to this PR.
```├─BROKEN─ https://api.github.com/repos/btcpayserver/btcpayserver/releases/latestServer (HTTP_404)├─BROKEN─ https://www.singhkays.com/blog/understanding-azure-b-series/ (ERRNO_ECONNREFUSED)```

**Comment by britttttk** [comment]
This is the only external mention I found looking in the other repos: [btcpayserver readme](https://github.com/btcpayserver/btcpayserver/blob/9ac5011bedcb8475cbaa9d9d38eae6f8eb3c21fd/README.md#features)

File: docs/WalletSetup.md [hunk 1]
```diff
@@ -23,10 +23,10 @@ Externally generated wallet can be used as:
 
 ### Use an existing hardware wallet
 
-Hardware wallets provide a good balance between security and ease of use. If you already have a hardware wallet set up, you can easily use it with your BTCPay Server. Thanks to built-in [hardware wallet integration](./Vault.md) the xpub key from the hardware wallet is automatically added to your BTCPay Server. The integration further allows you to spend funds received to your store within BTCPay's [internal wallet](./Wallet.md)
+Hardware wallets provide a good balance between security and ease of use. If you already have a hardware wallet set up, you can easily use it with your BTCPay Server. Thanks to built-in [hardware wallet integration](HardwareWalletIntegration.md) the xpub key from the hardware wallet is automatically added to your BTCPay Server. The integration further allows you to spend funds received to your store within BTCPay's [internal wallet](./Wallet.md)
```
Comment (britttttk) [inline docs/WalletSetup.md:26]:
```suggestion
Hardware wallets provide a good balance between security and ease of use. If you already have a hardware wallet set up, you can easily use it with your BTCPay Server. Thanks to built-in [hardware wallet integration](HardwareWalletIntegration.md) the xpub key from the hardware wallet is automatically added to your BTCPay Server. The integration further allows you to spend funds received to your store within BTCPay's [internal wallet](./Wallet.md).
```

Comment (britttttk) [reply docs/WalletSetup.md:26]: do we need `./HardwareWalletIntegration.md` here and below?

**Review by britttttk — COMMENTED** [review]
Can you add a :::warning::: that it only works for Bitcoin, not altcoins

File: docs/WalletSetup.md (hunk 1 above)
Comment (pavlenex) [reply docs/WalletSetup.md:26]: When I tested it locally it worked just fine, for some reason linkcheck provided an error for it so I switch it to that.

**Comment by pavlenex** [comment]
Great idea Brit. Added at the bottom of the page.
Capture

**Comment by pavlenex** [comment]
Thinking about the warning @britttttk can the way I formulated it mislead users into thinking HW that have altcoins aren't supported, let me know if it's confusing and please suggest if you have a better sentence in mind.
On the bright side test finally passing ✔️

**Comment by pavlenex** [comment]
I improved the warning, should be clearer now, if any complaints , please feel free to do a PR.
I'll merge this for now as it's something that's really important.

**Comment by Zaxounette** [comment]
w8

**Comment by Zaxounette** [comment]
> w8

Nevermind, I had a phrasing in mind but it doesn't do better than what you just changed.

**Pull request merged**